projects/libs/core-library-typescript/src/virtual-machine/hard-coded-contracts/registry/foobar-smart-contract.ts

```diff
@@ -5,12 +5,12 @@ export default class FooBarSmartContract extends BaseSmartContract {
 
   public async transfer(recipient: string, amount: number) {
     if (amount === 0) {
-      throw new Error("transaction amount must be > 0");
+      throw this.validationError("transaction amount must be > 0");
```

Comment:
Why not to use `new ValidationError(message)` here directly? Seems a bit more idiomatic.

Reply:
I want to control how the exception is being constructed, so a factory method would be a good way to contain the logic in one place.
It also keeps the concrete contract class unaware of the actual class representing the error.

Reply:
Ah, I see.